test/alternator/test_table.py

```diff
@@ -110,6 +110,16 @@ def test_create_and_delete_table_very_long_name(dynamodb):
 # Tests creating a table with an invalid schema should return a
 # ValidationException error.
 def test_create_table_invalid_schema(dynamodb):
+    # Creation of a table without AttributeDefinitions is illegal
+    with pytest.raises(ClientError, match='ValidationException.*[aA]ttributeDefinitions'):
+        dynamodb.create_table(
+            TableName='name_doesnt_matter',
+            BillingMode='PAY_PER_REQUEST',
+            KeySchema=[
+                { 'AttributeName': 'p', 'KeyType': 'HASH' },
+                { 'AttributeName': 'c', 'KeyType': 'RANGE' }
+            ],
+        )
```

Comment:
If you create a new issue, I propose that you don't modify the existing test function and instead create a new test function below it, with a comment explaining it reproduces this issue. 
As an example see test_create_table_duplicate_attribute_name,  test_create_table_invalid_attribute_definitions and the test I mentioned above test_create_table_spurious_attribute_definitions. All are specific error cases for AttributeDefinitions, each one was discovered in a different issue. So now you're adding another one. It can also be a new 5-line test function and doesn't need to be part of a bigger test.